vite_ruby/lib/tasks/vite.rake

```diff
@@ -4,6 +4,8 @@ $stdout.sync = true
 
 require 'rake'
 
+legacy_npm_version = `npm --version`.to_i < 7 rescue false
```

Comment:
@ElMassimo on second thought, do you think this is dangerous/undesirable to shell out to `npm` whenever `vite.rake` is loaded?

I wonder if this should be a proc instead and then it will only be executed within `if ARGV.include?('assets:precompile')` and within `task :install_dependencies`.

In other words:

```ruby
legacy_npm_version = -> { `npm --version`.to_i < 7 rescue false }

# ...

if ARGV.include?('assets:precompile')
  if legacy_npm_version.call
    # ...
```

Reply:
Not dangerous, but probably undesirable, yes, as the rake task will be loaded by default when the gem is loaded.

I'll make the suggested change to reuse the logic and avoid the shell execution, before releasing 👍

Reply:
Great, thanks!

Reply:
Released in https://github.com/ElMassimo/vite_ruby/releases/tag/vite_ruby@3.1.7. Thanks!